sssd_test_framework/utils/local_override.py

```diff
@@ -0,0 +1,414 @@
+        :type name: str
+        :return: New user local override object.
+        :rtype: LocalOverrideUser
+        """
+        return LocalOverrideUser(self, name)
+
+    def group(self, group: str) -> LocalOverrideGroup:
+        """
+        Get local override group object.
+
+        .. code-block:: python
+            :caption: Example usage
+
+            @pytest.mark.topology(KnownTopology.Client)
+            def test_example(client: Client):
+                # Create local override
+                client.local_override.group("users").add(name="o-users", gid=10001)
+
+                # Assert the overridden name and gid
+                assert client.tools.getent.group("o-users").gid == 10001
+
+        :param group: Group name.
+        :type group: str
+        :return: New group object.
+        :rtype: LocalOverrideGroup
+        """
+        return LocalOverrideGroup(self, group)
+
+
+class LocalOverrideUser:
+    """
+    Management of local override users.
+    """
+
+    def __init__(self, util: LocalOverrideUtils, user: str) -> None:
+        """
+        :param util: OverrideUser object.
+        :type util: LocalOverrideUser
+        :param user: User name.
+        :type user: str
+        """
+        self.util: LocalOverrideUtils = util
+        self.user: str = user
+
+    def add(
+        self,
+        *,
+        uid: int | None = None,
+        gid: int | None = None,
+        name: str | None = None,
+        home: str | None = None,
+        gecos: str | None = None,
+        shell: str | None = None,
+        certificate: str | None = None,
+    ) -> LocalOverrideUser:
+        """
+        Create new local override user.
+
+        :param uid: User id, defaults to None
+        :type uid: int | None, optional
+        :param gid: Primary group id, defaults to None
+        :type gid: int | None, optional
+        :param name: | None, optional
+        :type name: str | None, optional
+        :param home: Home directory, defaults to None
+        :type home: str | None, optional
+        :param gecos: GECOS, defaults to None
+        :type gecos: str | None, optional
+        :param shell: Login shell, defaults to None
+        :type shell: str | None, optional
+        :param certificate: PKI certificate defaults to None
+        :type certificate: str | None, optional
+        :return: Self.
+        :rtype: LocalOverrideUser
+        """
+        args: CLIBuilderArgs = {
+            "name": (self.util.cli.option.VALUE, name),
+            "uid": (self.util.cli.option.VALUE, uid),
+            "gid": (self.util.cli.option.VALUE, gid),
+            "home": (self.util.cli.option.VALUE, home),
+            "gecos": (self.util.cli.option.VALUE, gecos),
+            "shell": (self.util.cli.option.VALUE, shell),
+            "certificate": (self.util.cli.option.VALUE, certificate),
+        }
+
+        self.util.logger.info(f"Creating local override user {self.user} on {self.util.host.hostname}")
```

Comment:
user local override
(or local override for user, whatever is better)

Reply:
Same on other places.

Because you are not creating new user, you are creating the override. "local override user" suggest that you create new user object.